### PR TITLE
updated fleet registration for openshift cluster

### PR DESCRIPTION
**Pull request by aryan16**
Added an extra parameter while creating membership for openshift cluster.

### CODE REVIEW

File: asmcli/lib/installation_dependencies.sh [hunk 1]
```diff
@@ -43,7 +43,8 @@ required_iam_roles() {
 }
 # [END required_iam_roles]
 
-old_required_apis() {
+# [START required_apis]
+required_apis() {
```
Comment (ZhengzheYang) [inline asmcli/lib/installation_dependencies.sh:47]: is there a reason for the function name change? I think this was changed in #1152 for a reason, but I am not sure.

File: asmcli/lib/installation_dependencies.sh [hunk 2]
```diff
@@ -66,20 +67,6 @@ EOF
...
-  local CA; CA="$(context_get-option "CA")"
-    cat << EOF
-mesh.googleapis.com
-EOF
-  case "${CA}" in
-   gcp_cas)
-     echo privateca.googleapis.com
-     ;;
-    *);;
-  esac
```
Comment (ZhengzheYang) [inline asmcli/lib/installation_dependencies.sh:82]: same here. Is there a reason we delete these lines?

File: asmcli/lib/installation_dependencies.sh [hunk 3]
```diff
@@ -409,4 +400,4 @@ get_auth_token() {
   local PROJECT_ID; PROJECT_ID="$(context_get-option "PROJECT_ID")"
   local TOKEN; TOKEN="$(retry 2 gcloud --project="${PROJECT_ID}" auth print-access-token)"
   echo "${TOKEN}"
-}
+}
```
Comment (ZhengzheYang) [inline asmcli/lib/installation_dependencies.sh:403]: new line

File: asmcli/lib/installation_dependencies.sh [hunk 4]
```diff
@@ -320,6 +307,7 @@ register_cluster() {
   local FLEET_ID; FLEET_ID="$(context_get-option "FLEET_ID")"
   local KCF; KCF="$(context_get-option "KUBECONFIG")"
   local KCC; KCC="$(context_get-option "CONTEXT")"
+  local OPENSHIFT_CLUSTER; OPENSHIFT_CLUSTER="$(context_get-option "OPENSHIFT_CLUSTER_SETUP")"
```
Comment (ZhengzheYang) [inline asmcli/lib/installation_dependencies.sh:310]: I don't see where you set this variable. You might need to call `context_set-option` to set it somewhere and maybe add it to the [context file](https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/blob/main/asmcli/lib/context.sh#L68) to get a default env var value.

File: asmcli/lib/installation_dependencies.sh (hunk 1 above)
Comment (aryan16) [reply asmcli/lib/installation_dependencies.sh:47]: I didn't change this, it was required to run `scripts/release-asm/precommit` for the PR

File: asmcli/lib/installation_dependencies.sh (hunk 2 above)
Comment (aryan16) [reply asmcli/lib/installation_dependencies.sh:82]: Same reason as above.

File: asmcli/lib/installation_dependencies.sh (hunk 1 above)
Comment (ZhengzheYang) [reply asmcli/lib/installation_dependencies.sh:47]: the precommit only merges the changes made by you so maybe you changed this by accident or have a stale commit. I would suggest keeping only the changes made by you and rebase on HEAD.

**Pull request closed without merging**